### PR TITLE
update: added log to notation login and logout commands

### PR DESCRIPTION
**Pull request by patrickzheng200**
This PR adds logs to the login and logout commands.

This PR is a part of resolving https://github.com/notaryproject/roadmap/issues/71.

### CODE REVIEW

File: cmd/notation/login.go [hunk 1]
```diff
@@ -47,23 +50,28 @@ Example - Login using $NOTATION_USERNAME $NOTATION_PASSWORD variables:
...
 		},
 		RunE: func(cmd *cobra.Command, args []string) error {
-			return runLogin(cmd, opts)
+			return runLogin(cmd.Context(), opts)
 		},
 	}
+	opts.LoggingFlagOpts.ApplyFlags(command.Flags())
 	command.Flags().BoolVar(&opts.passwordStdin, "password-stdin", false, "take the password from stdin")
-	opts.ApplyFlags(command.Flags())
+	opts.SecureFlagOpts.ApplyFlags(command.Flags())
```
Comment (JeyJeyGao) [inline cmd/notation/login.go:58]: Can we swap line 58 and 57

Comment (patrickzheng200) [reply cmd/notation/login.go:57]: Updated.

**Review by priteshbandi — COMMENTED** [review]
Looks like we are not adding much information in debug logs, do we really want debug log for login logout commands?

File: cmd/notation/login.go [hunk 2]
```diff
@@ -47,23 +50,28 @@ Example - Login using $NOTATION_USERNAME $NOTATION_PASSWORD variables:
 			return nil
 		},
 		RunE: func(cmd *cobra.Command, args []string) error {
-			return runLogin(cmd, opts)
+			return runLogin(cmd.Context(), opts)
 		},
 	}
+	opts.LoggingFlagOpts.ApplyFlags(command.Flags())
+	opts.SecureFlagOpts.ApplyFlags(command.Flags())
```
[the remark this answers is not in the document]
Comment (patrickzheng200) [reply cmd/notation/login.go:57]: > why do we need `SecureFlagOpts` for login/logout?

This flag only appears in login. It contains username and password fields which are required in the login process.

**Comment by patrickzheng200** [comment]
> Looks like we are not adding much information in debug logs, do we really want debug log for login logout commands?

Oh yeah, so I think one good reason to add debug logs to login is that there are several https roundtrips among notation, registry, and authorization service -> [spec](https://github.com/notaryproject/notation/blob/main/specs/registry-auth.md). Printing out those traces could be useful during debug.
As for logout, the logger prints out the name of the credential helper program for debug purpose, as we are using Docker's credential helpers.

File: pkg/auth/native_store.go [hunk 3]
```diff
@@ -25,22 +27,24 @@ type nativeAuthStore struct {
...
 }
 
 // newNativeAuthStore creates a new native store that uses a remote helper
 // program to manage credentials. Note: it's different from the nativeStore in
 // docker-cli which may fall back to plain text store
-func newNativeAuthStore(helperSuffix string) CredentialStore {
+func newNativeAuthStore(ctx context.Context, helperSuffix string) CredentialStore {
+	logger := log.GetLogger(ctx)
 	name := remoteCredentialsPrefix + helperSuffix
+	logger.Infoln("Executing remote credential helper program :", name)
```
Comment (shizhMSFT) [inline pkg/auth/native_store.go:47]: nit:
```suggestion
	logger.Infoln("Executing remote credential helper program:", name)
```

**Review by shizhMSFT — APPROVED** [review]
LGTM with suggestions.

File: specs/commandline/logout.md [hunk 4]
```diff
@@ -23,9 +25,3 @@ Flags:
 ```shell
 notation logout registry.example.com
 ```
-
-### Log out from all logged in registries
-
-```shell
-notation logout --all
-```
```
Comment (priteshbandi) [inline specs/commandline/logout.md:31]: Lets open an issue to track this #

**Review by priteshbandi — APPROVED** [review]
LGTM

**Pull request merged**